### PR TITLE
Fixes CookieTempDataProvider to handle empty request PathBase while s…

### PR DESCRIPTION
**Pull request by kichalla**
…etting cookie options

@Eilon @pranavkm @dougbu 

Fixes: https://github.com/aspnet/Mvc/issues/5512

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/CookieTempDataProvider.cs [hunk 1]
```diff
@@ -63,7 +63,7 @@ public void SaveTempData(HttpContext context, IDictionary<string, object> values
 
             var cookieOptions = new CookieOptions()
             {
-                Path = string.IsNullOrEmpty(_options.Value.Path) ? context.Request.PathBase.ToString() : _options.Value.Path,
+                Path = string.IsNullOrEmpty(_options.Value.Path) ? GetPathBase(context) : _options.Value.Path,
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/CookieTempDataProvider.cs]: Can you write a comment that says why an empty string is problematic?

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/TempDataInCookiesTest.cs [hunk 2]
```diff
@@ -37,29 +39,101 @@ public TempDataInCookiesTest(MvcTestFixture<BasicWebSite.StartupWithCookieTempDa
...
-            var cookies = response.Headers.GetValues(HeaderNames.SetCookie);
-            var cookieTempDataProviderCookies = cookies.Where(cookie => cookie.Contains(CookieTempDataProvider.CookieName));
-            Assert.NotNull(cookieTempDataProviderCookies);
+            IEnumerable<string> setCookieValues;
+            Assert.True(response.Headers.TryGetValues(HeaderNames.SetCookie, out setCookieValues));
+            setCookieValues = setCookieValues.Where(cookie => cookie.Contains(CookieTempDataProvider.CookieName));
 
             // Verify that all the cookies from CookieTempDataProvider are within the maximum size
-            foreach (var cookie in cookieTempDataProviderCookies)
+            foreach (var cookie in setCookieValues)
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/TempDataInCookiesTest.cs]: `Assert.NotEmpty(setCookieValue);`

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/TempDataInCookiesTest.cs [hunk 3]
```diff
@@ -37,29 +39,101 @@ public TempDataInCookiesTest(MvcTestFixture<BasicWebSite.StartupWithCookieTempDa
...
             {
                 Assert.True(cookie.Length <= ChunkingCookieManager.DefaultChunkSize);
             }
 
+            var cookieTempDataProviderCookies = setCookieValues
+                .Select(setCookieValue => SetCookieHeaderValue.Parse(setCookieValue))
+                .Where(setCookieHeader => setCookieHeader.Name == CookieTempDataProvider.CookieName);
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/TempDataInCookiesTest.cs]: Why do you need to filter these? Aren't they already filtered by `TryGetValue`?

**Comment by kichalla** [comment]
@pranavkm 🆙 📅

**Comment by kichalla** [comment]
a3c06b00ccc5eca95b38b70e4b22ab4f29527242

**Pull request closed without merging**